app/src/main/java/com/example/henzoshimada/feeltrip/EditMoodActivity.java

```diff
@@ -184,69 +177,94 @@ protected void onCreate(Bundle savedInstanceState) {
                     ImageView imageView = (ImageView) findViewById(R.id.imgView);
                     imageView.setImageBitmap(decodedPhoto);
+                    TextView modeLPhotoText = (TextView) findViewById(R.id.modePhoto);
+                    modeLPhotoText.setText("On");
+                    modeLPhotoText.setTextColor(getResources().getColor(R.color.green));
                 }
             } catch (Exception e) {
                 Log.d("tag", "No image found in editmood");
             }
             try {
                 latitude = editmood.getLatitude();
-            }
-            catch (Exception e) {
+            } catch (Exception e) {
                 Log.d("tag", "No latitude found in editmood");
             }
             try {
                 longitude = editmood.getLongitude();
-            }
-            catch (Exception e) {
+                TextView modeLocationText = (TextView) findViewById(R.id.modeLocation);
+                modeLocationText.setText("On");
+                modeLocationText.setTextColor(getResources().getColor(R.color.green));
+                locationOn = true;
+            } catch (Exception e) {
                 Log.d("tag", "No longitude found in editmood");
             }
             try {
                 inputMoodDescription.setText(editmood.getDescription());
-            }
-            catch (Exception e) {
+            } catch (Exception e) {
                 Log.d("tag", "No description found in editmood");
             }
             try {
-                socialSit = editmood.getSocialSit(); //TODO: There is a massive UI bug where if editmood's privacy settings are currently set to public, and the user then changes the socialsit, it changes the privacy settings too.
-            }
-            catch (Exception e) {
+                socialSit = editmood.getSocialSit();
+            } catch (Exception e) {
                 Log.d("tag", "No socialsit found in editmood");
             }
             try {
                 emotionalState = editmood.getEmotionalState();
-            }
-            catch (Exception e) {
+                TextView emotionString = (TextView) findViewById(R.id.emotionString);
+                emotionString.setText(emotionalState);
+                emotionString.setTextColor(getResources().getColor(R.color.green));
+            } catch (Exception e) {
                 Log.d("tag", "No emotionalstate found in editmood");
             }
-
+            try {
+                emojiID = editmood.getEmoji();
+            } catch (Exception e) {
+                Log.d("tag", "No emoji found in editmood");
+            }
+            if (!editmood.getPrivate()) {
+                TextView modePostText = (TextView) findViewById(R.id.modePost);
+                modePostText.setText(R.string.mode_public);
+                modePostText.setTextColor(getResources().getColor(R.color.green));
+                showPublicOn = true;
```

Comment:
This block of code was missing making editing privacy settings impossible. This has been fixed.